### PR TITLE
feat: Check listing prices of nfts when loading confirm modal

### LINKED ISSUE #8889: Check listing prices of nfts when loading confirm modal
yesterday, it happened to me, that I was buying some nfts, but it loaded the wrong prices for me. The price had already changed but i did not refresh. So, after opening the confirm modal - with old prices, I clicked buy and threw an error about reserve price. 

can we also check listing prices when loading the modal? The load modal currently waits for the funds check IIRC.

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Needs QA check

- @kodadot/qa-guild please review

- pls check the following scenarios 

nft price updated
-  quick buy -> notification "NFT price updated"
-  cart -> notification "NFT price updated" if one item , if more notification "NFT {nftName} price updated"

nft not listed anymore, (someone bought that item), that item will be removed , and if nft count is 0 the modal will be closed 

-  quick buy  -> notification "NFT is not listed anymore" and modal close
-  cart -> if one item notification "NFT is not listed anymore" and modal close . if more items  -> notification "NFT {nftName} is not listed anymore"

## Context

- [x] Closes #8889
- [x] Related #8491

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI;

### CODE REVIEW

File: components/buy/Buy.vue [hunk 1]
```diff
@@ -21,16 +22,26 @@ import { ShoppingCartItem } from '@/components/common/shoppingCart/types'
 import { AutoTeleportActionButtonConfirmEvent } from '@/components/common/autoTeleport/AutoTeleportActionButton.vue'
 import { warningMessage } from '@/utils/notification'
 import type { AutoTeleportAction } from '@/composables/autoTeleport/types'
+import _ from 'lodash'
```
Comment (Jarsen136) [inline components/buy/Buy.vue:25]: import isEqual from 'lodash/isEqual'

**Review by Jarsen136 — APPROVED** [review]

**Comment by prury** [comment]
@hassnian 
was it supposed to work here?

https://github.com/kodadot/nft-gallery/assets/36627808/abfb06e5-26c1-4758-a68f-f7923a7efb97

**Comment by hassnian** [comment]
@prury works for me  , pls try again 

ahp
https://github.com/kodadot/nft-gallery/assets/44554284/3c28f749-4ee7-4784-bc88-bdbd32b44dc6

ahk , messed up second try , still worked
https://github.com/kodadot/nft-gallery/assets/44554284/78597621-360c-422b-85f4-3fcbb8c2baed

**Comment by prury** [comment]
Delisted and tried to buy✔️ 
https://github.com/kodadot/nft-gallery/assets/36627808/5ead9081-a40a-4f11-8b69-ac84d478e511

changed price and tried to buy ✔️ 
https://github.com/kodadot/nft-gallery/assets/36627808/1a6e0dd8-ba16-4b8c-8855-16c65a5f5330

bought with a third account and tried to buy ❌ 
https://github.com/kodadot/nft-gallery/assets/36627808/eba3a736-4f97-4263-a400-00031935f969

don't mind the values on explorer, its a bug

will test cart interactions on round 2

**Review by preschian — APPROVED** [review]

**Comment by hassnian** [comment]
@prury  pls try again 

https://github.com/kodadot/nft-gallery/assets/44554284/4d900366-032e-447a-afb4-a44b91a032ee

**Comment by prury** [comment]
cool, tried both cart interactions and it went well

cart buy and cart change price: ✔️ 
https://github.com/kodadot/nft-gallery/assets/36627808/2c2e7e54-ab11-496d-bb0d-b6b4d712eb5b

cart normal buy round 2: ✔️ 
https://github.com/kodadot/nft-gallery/assets/36627808/a517304e-a2fb-47a5-b101-51b7053f9f9a

well done @hassnian!

**Comment by yangwao** [comment]
Thanks!
pay 50 usd

**Pull request merged**